preprocess/mmhsum_main.cc

```diff
@@ -1,15 +1,28 @@
 #include "util/murmur_hash.hh"
 
 #include <iostream>
+#include <cstring>
 
 int main(int argc, char *argv[]) {
   if (argc > 1) {
     std::cerr << "Usage: [stdin] " << argv[0] << std::endl;
     return 1;
   }
-  std::string input;
-  for (std::string line; std::getline(std::cin, line);) {
-    input += line;
+  
+  constexpr size_t bufferSize = 1024*1024;
+  char* buffer = new char[bufferSize];
+  uint64_t chained_hash = 0;
+  
+  while (std::cin)
+  {
+    std::cin.read(buffer, bufferSize);
+    size_t count = std::cin.gcount();
+    if (!count)
+      break;
+    if (std::cin.eof())
+      count -= 1;
```

Comment:
Is that a thing?  Does std::cin actually add a character to the read bytes at the end of a file?

Reply:
Nope, my mistake. I was comparing results with the previous code and I though it was that, but it was simply the endline `echo` does by default. Previous code was ignoring that endline because `getline` was used instead of `read`.